core/include/byteorder.h

```diff
@@ -460,12 +485,28 @@ static inline uint16_t byteorder_bebuftohs(const uint8_t *buf)
     return (uint16_t)((buf[0] << 8) | (buf[1] << 0));
 }
 
+static inline uint32_t byteorder_bebuftohl(const uint8_t *buf)
+{
+    return (uint32_t)((buf[0] << 24) |
+                      (buf[1] << 16) |
+                      (buf[2] << 8) |
+                      (buf[3] << 0));
```

Comment:
couldn't resist the nitpicking

```suggestion
    return (uint32_t)((buf[0] << 24)
                    | (buf[1] << 16)
                    | (buf[2] << 8)
                    | (buf[3] << 0));
```

would look a bit nicer IMHO